pkg/testrunner/helper.go

```diff
@@ -0,0 +1,68 @@
+package testrunner
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/Masterminds/semver"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getLatestK8sVersion(gardenerKubeconfigPath, cloudprofile, cloudprovider string) (string, error) {
+	ctx := context.Background()
+	defer ctx.Done()
+	k8sGardenClient, err := kubernetes.NewClientFromFile(gardenerKubeconfigPath, nil, client.Options{
+		Scheme: kubernetes.GardenScheme,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	profile := &gardenv1beta1.CloudProfile{}
+	err = k8sGardenClient.Client().Get(ctx, types.NamespacedName{Name: cloudprofile}, profile)
+	if err != nil {
+		return "", err
+	}
+
+	rawVersions, err := getCloudproviderVersions(profile, cloudprovider)
+	if err != nil {
+		return "", err
+	}
+
+	if len(rawVersions) == 0 {
+		return "", fmt.Errorf("No kubernetes versions found for cloudprofle %s", cloudprofile)
+	}
+
+	versions := make([]*semver.Version, len(rawVersions))
+	for i, rawVersion := range rawVersions {
+		v, err := semver.NewVersion(rawVersion)
+		if err == nil {
+			versions[i] = v
+		}
+	}
+	sort.Sort(semver.Collection(versions))
+
+	return versions[len(versions)-1].String(), nil
+}
+
+func getCloudproviderVersions(profile *gardenv1beta1.CloudProfile, cloudprovider string) ([]string, error) {
+	switch gardenv1beta1.CloudProvider(cloudprovider) {
```

Comment:
Is it possible to fetch the version information in a "generic" way? One where we don't have to adapt the testrunner once a new provider is added?

Reply:
unfortunately not because we need a mapping from a gostruct field to a string.

I think it could be generic by using reflection, but this has a bad smell and I think we should be consistent with the gardener style (https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/cloudbotanist.go#L46)